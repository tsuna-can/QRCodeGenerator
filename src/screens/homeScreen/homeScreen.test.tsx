import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from './homeScreen';

const navigation = {
  navigate: jest.fn(),
};

const route = {
  params: {
    fixedValue: '',
    variableValue: '',
    digits: 1,
  },
};

describe('HomeScreen', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<HomeScreen navigation={navigation} route={route} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
