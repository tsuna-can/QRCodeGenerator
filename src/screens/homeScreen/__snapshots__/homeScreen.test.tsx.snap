// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            style={
              {
                "flex": 1,
                "margin": 10,
              }
            }
          >
            <View
              style={
                {
                  "flex": 10,
                }
              }
            >
              <RCTScrollView>
                <View>
                  <View
                    style={
                      {
                        "flex": 6,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "marginVertical": 10,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "black",
                          }
                        }
                      >
                        Fixed Part
                      </Text>
                      <TextInput
                        autoCapitalize="none"
                        autoCompleteType="email"
                        label="Fixed Part"
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        placeholder="Fixed Part"
                        style={
                          {
                            "backgroundColor": "white",
                            "borderRadius": 10,
                            "marginVertical": 10,
                            "paddingLeft": 10,
                          }
                        }
                        value=""
                      />
                      <Text
                        style={
                          {
                            "color": "red",
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        {
                          "marginVertical": 10,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "black",
                          }
                        }
                      >
                        Variable Part
                      </Text>
                      <TextInput
                        autoCapitalize="none"
                        autoCompleteType="number"
                        keyboardType="numeric"
                        label="Variable Part"
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        placeholder="Variable Part"
                        style={
                          {
                            "backgroundColor": "white",
                            "borderRadius": 10,
                            "marginVertical": 10,
                            "paddingLeft": 10,
                          }
                        }
                        value=""
                      />
                      <Text
                        style={
                          {
                            "color": "red",
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        {
                          "flex": 1,
                          "marginVertical": 10,
                          "zIndex": 1,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "black",
                          }
                        }
                      >
                        Digits for variable part
                      </Text>
                      <View
                        style={
                          [
                            {
                              "width": "100%",
                            },
                            {
                              "zIndex": 5000,
                            },
                            {},
                          ]
                        }
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": false,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onLayout={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "#fff",
                              "borderColor": "#000",
                              "borderRadius": 8,
                              "borderWidth": 0,
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "marginVertical": 10,
                              "minHeight": 50,
                              "opacity": 1,
                              "paddingHorizontal": 10,
                              "paddingVertical": 3,
                              "width": "100%",
                              "zIndex": 5000,
                            }
                          }
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#000",
                                  "flex": 1,
                                },
                                {},
                                false,
                                {},
                              ]
                            }
                          >
                            Select an item
                          </Text>
                          <View
                            style={
                              [
                                {
                                  "marginLeft": 10,
                                },
                                {},
                              ]
                            }
                          >
                            <Image
                              source={
                                {
                                  "testUri": "../../react-native-dropdown-picker/src/themes/light/icons/arrow-down.png",
                                }
                              }
                              style={
                                [
                                  {
                                    "height": 20,
                                    "width": 20,
                                  },
                                  {},
                                ]
                              }
                            />
                          </View>
                        </View>
                      </View>
                      <View>
                        <Text
                          style={
                            {
                              "color": "red",
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          {
                            "borderRadius": 2,
                          },
                          {
                            "flex": 1,
                            "marginHorizontal": 5,
                            "marginVertical": 5,
                          },
                          false,
                        ]
                      }
                      testID="RNE_BUTTON_WRAPPER"
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          {
                            "busy": false,
                            "checked": undefined,
                            "disabled": false,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "alignSelf": "flex-end",
                              "backgroundColor": "#ffffff",
                              "borderColor": "#2089dc",
                              "borderRadius": 100,
                              "borderWidth": 0,
                              "elevation": 1,
                              "flexDirection": "row-reverse",
                              "height": 50,
                              "justifyContent": "center",
                              "padding": 8,
                              "paddingHorizontal": 10,
                              "width": 50,
                            }
                          }
                        >
                          <View
                            style={
                              {
                                "overflow": "hidden",
                              }
                            }
                            testID="RNE__ICON__CONTAINER"
                          >
                            <View
                              accessibilityRole="button"
                              android_ripple={
                                {
                                  "borderless": false,
                                  "color": "rgba(0, 0, 0, 0.3)",
                                  "radius": -5,
                                }
                              }
                              disabled={false}
                              testID="RNE__ICON__CONTAINER_ACTION"
                            >
                              <View
                                style={
                                  {
                                    "alignItems": "center",
                                    "backgroundColor": "transparent",
                                    "justifyContent": "center",
                                  }
                                }
                                testID="RNE__ICON"
                              >
                                <Text
                                  allowFontScaling={false}
                                  selectable={false}
                                  style={
                                    [
                                      {
                                        "color": "orange",
                                        "fontSize": 24,
                                      },
                                      {
                                        "backgroundColor": "transparent",
                                      },
                                      {
                                        "fontFamily": "Material Design Icons",
                                        "fontStyle": "normal",
                                        "fontWeight": "normal",
                                      },
                                      {},
                                    ]
                                  }
                                  testID="RNE__ICON__Component"
                                >
                                  󰓎
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flex": 3,
                        "justifyContent": "center",
                      }
                    }
                  />
                  <View
                    style={
                      {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          {
                            "borderRadius": 2,
                          },
                          {
                            "flex": 1,
                            "marginHorizontal": 5,
                            "marginVertical": 5,
                          },
                          false,
                        ]
                      }
                      testID="RNE_BUTTON_WRAPPER"
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          {
                            "busy": false,
                            "checked": undefined,
                            "disabled": false,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "alignSelf": "flex-end",
                              "backgroundColor": "#ffffff",
                              "borderColor": "#2089dc",
                              "borderRadius": 100,
                              "borderWidth": 0,
                              "elevation": 1,
                              "flexDirection": "row-reverse",
                              "height": 50,
                              "justifyContent": "center",
                              "padding": 8,
                              "paddingHorizontal": 10,
                              "width": 50,
                            }
                          }
                        >
                          <View
                            style={
                              {
                                "overflow": "hidden",
                              }
                            }
                            testID="RNE__ICON__CONTAINER"
                          >
                            <View
                              accessibilityRole="button"
                              android_ripple={
                                {
                                  "borderless": false,
                                  "color": "rgba(0, 0, 0, 0.3)",
                                  "radius": -5,
                                }
                              }
                              disabled={false}
                              testID="RNE__ICON__CONTAINER_ACTION"
                            >
                              <View
                                style={
                                  {
                                    "alignItems": "center",
                                    "backgroundColor": "transparent",
                                    "justifyContent": "center",
                                  }
                                }
                                testID="RNE__ICON"
                              >
                                <Text
                                  allowFontScaling={false}
                                  selectable={false}
                                  style={
                                    [
                                      {
                                        "color": "black",
                                        "fontSize": 24,
                                      },
                                      {
                                        "backgroundColor": "transparent",
                                      },
                                      {
                                        "fontFamily": "Material Icons",
                                        "fontStyle": "normal",
                                        "fontWeight": "normal",
                                      },
                                      {},
                                    ]
                                  }
                                  testID="RNE__ICON__Component"
                                >
                                  
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderRadius": 2,
                    },
                    {
                      "flex": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 5,
                    },
                    false,
                  ]
                }
                testID="RNE_BUTTON_WRAPPER"
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": false,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#2089dc",
                        "borderColor": "#2089dc",
                        "borderRadius": 5,
                        "borderWidth": 0,
                        "flex": 1,
                        "flexDirection": "row-reverse",
                        "justifyContent": "center",
                        "padding": 8,
                        "paddingHorizontal": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "white",
                          "fontSize": 18,
                          "paddingVertical": 1,
                          "textAlign": "center",
                        }
                      }
                    >
                      Start Generate
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              onLayout={[Function]}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": [
                    {
                      "translateY": 0,
                    },
                  ],
                }
              }
              testID="toastAnimatedContainer"
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "#FFF",
                    "borderLeftColor": "#69C779",
                    "borderLeftWidth": 5,
                    "borderRadius": 6,
                    "elevation": 2,
                    "flexDirection": "row",
                    "height": 60,
                    "opacity": 1,
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 6,
                    "width": 340,
                  }
                }
                testID="toastTouchableContainer"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "flex-start",
                        "flex": 1,
                        "justifyContent": "center",
                        "paddingHorizontal": 25,
                      },
                      undefined,
                    ]
                  }
                  testID="toastContentContainer"
                />
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
